### ISSUE TITLE
DPO Qwen2.5VL issue

### ISSUE THREAD

**Issue by aaaalonga** [modelscope/ms-swift#4940]
Thanks for this wonderful framework, and I have following questions：
1. I want all the input images to be 280*280 in size, where do I need to modify？
2. I need to customize a loss function, but I see that the documentation says that the plugin doesn't support DPO. Where do I need to modify？

**Comment by hjh0119** [comment]
1. Use the environment variable MAX_PIXELS (only effective for QwenVL), or the parameter --max_pixels.
2. In DPOTrainer, override the compute_loss method (currently, the one inherited from HFDPOTrainer is used).

**Comment by aaaalonga** [comment]
Thanks for your answer and sorry to bother you again. When I try to override computer_loss method, I find that ['chosen_pixel_values'] and ['chosen_pixel_values'] are in the train_dataset (I get from SwiftSft.run(), a LazyLLMDataset class ). And I have another two questions:
1. I have set MAX_PIXELS=78400 and MIN_PIXELS=78400 (280 * 280), but the shape  of ['chosen_pixel_values'] and ['chosen_pixel_values'] is torch[364, 1176] not torch[400, 1176]. And  ['chosen_image_grid_thw'] is torch([[1, 14, 26]]) not torch([[1, 20, 20]]). It seems MAX_PIXELS and MIN_PIXELS can not work together.
2.In my custom losses, one of losses is constructed by accept image and reject image. So can I make ['chosen_pixel_values'] and ['chosen_pixel_values'] different through dataset resgister.

**Comment by hjh0119** [comment]
1. The logic of the environment variable `max_pixels` is to set model.processor, so the image processing logic is inside the processor. The relevant code can be found here: https://github.com/modelscope/ms-swift/blob/v3.6.1/swift/llm/model/model/qwen.py#L563-L574

2. The logic for handling rejected images has not been implemented yet; I will implement it soon.
